pkg/backend/filestate/backend.go

```diff
@@ -550,11 +551,58 @@ func (b *localBackend) apply(
 
 // query executes a query program against the resource outputs of a locally hosted stack.
 func (b *localBackend) query(ctx context.Context, op backend.QueryOperation,
-	events chan<- engine.Event) result.Result {
+	callerEventsOpt chan<- engine.Event) result.Result {
```

Comment:
All of this code could be shared with the cloud backend, but I'm not entirely sure where to put it.

Reply:
`pkg/backend/query`, perhaps?

Reply:
it seems like a mild layer violation to put it there, but that's fine I guess.